### PR TITLE
MGDAPI-4390 - Move rhsso to the productised Go operator

### PR DESCRIPTION
**Pull request by valerymo**
# Issue link
https://issues.redhat.com/browse/MGDAPI-4390

## Notes
- PR is intended for release **1.26**
- PR does not contain 1.26.0 rhoam bundle that required for verification (release management related)
- Testing Branch **1.25.1** was used for verification (as rhoam bundle is not included) - it's a copy of the PR branch, with rhoam bundle: https://github.com/valerymo/integreatly-operator/tree/1.25.1
- New RHSSO operator 7.6 used in 1.25.1 CSV: https://github.com/valerymo/integreatly-operator/blob/1.25.1/bundles/managed-api-service/1.25.1/manifests/managed-api-service.clusterserviceversion.yaml#L63 . Similar need to be done while adding 1.26.0 bundle.

# What
The community operator is deprecated and soon to become unsupported.
We need to move to the productised version of the operator.

# How it was verified
*  *Pipeline*: Copy of Nightly Pipeline `managed-api-upgrade` was used for verification: https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/blue/organizations/jenkins/Playground%2Fvalerym%2Fmanaged-api-upgrade-4390/detail/managed-api-upgrade-4390/21/pipeline
** 1.25.1 branch was used for verification in Pipeline (please see Notes for more details)
** Tests produced in pipeline
* OLM local (from operator hub) Install & Upgrade - done from 1.25.0 to 1.25.1, - keycloack 18.0 was updated to RHSSO 7.6.
** H07 and B01B tests passed locally

# Verification steps
- Verify that all e2e tests pass with the product operator
- Verify that the fake version transition upgrade is feasible
- General upgrade testing is critical. Verify all e2e tests pass after upgrade.

### CODE REVIEW

File: bundles/rhsso-operator/bundles.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+bundles:
+    - name: keycloak-operator.v18.0.0
+      image: quay.io/integreatly/rhsso-bundle:18.0.0
+    - name: rhsso-operator.7.5.2-9
+      image: quay.io/integreatly/rhsso-bundle:7.5.2-9
+    - name: rhsso-operator.7.6.0-4
+      image: quay.io/integreatly/rhsso-bundle:7.6.0-4
```
Comment (MStokluska) [inline bundles/rhsso-operator/bundles.yaml:7]: @valerymo should this not be the official registry bundle instead?

File: manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,388 @@
+apiVersion: operators.coreos.com/v1alpha1
```
Comment (MStokluska) [inline manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml:1]: These files should not be here.
The bundle consists of a manifests folder - CSV and CRDs etc and a metadata folder - with annotations file.

File: manifests/integreatly-rhsso/7.6.0-4/keycloakbackups.keycloak.org.crd.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (MStokluska) [inline manifests/integreatly-rhsso/7.6.0-4/keycloakbackups.keycloak.org.crd.yaml:1]: Same comment as above; there should a manifests and annotations folders here.

File: manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (MStokluska) [reply manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml:1]: Actually no; let's do it same way as it's done for 3scale.
So the actualy files in here are only the manifests; no bundle dockerfile and manifests and annotations folders please.

For reference - this is how it's done for 3scale (or CRO) https://github.com/integr8ly/integreatly-operator/tree/master/manifests/integreatly-3scale/0.9.0

**Comment by MStokluska** [comment]
@valerymo I think to satisfy the ci/prow/manifest I would drop the 18.0.0 manifests entirely and remove the replaces in the next version CSV.

File: scripts/create-rhsso-operator-index.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
```
Comment (MStokluska) [inline scripts/create-rhsso-operator-index.sh:1]: Would it not be better to re-use 3scale script to generate indexes but make it generic enough to support both?

File: manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (valerymo) [reply manifests/integreatly-rhsso/7.5.2-9/rhsso-operator.clusterserviceversion.yaml:1]: @MStokluska  Thank you very much for comment. It was mistake, only in 7.5.2-9, other -18.0 and 7.6.0-4. 
Done. Thanks!

File: scripts/create-rhsso-operator-index.sh (hunk 4 above)
Comment (valerymo) [reply scripts/create-rhsso-operator-index.sh:1]: Jira opened for discussion https://issues.redhat.com/browse/MGDAPI-4481 cc @MStokluska @briangallagher

File: bundles/rhsso-operator/bundles.yaml (hunk 1 above)
Comment (valerymo) [reply bundles/rhsso-operator/bundles.yaml:7]: @MStokluska Thanks very much for comment. I will try use official bundle for 7.6.0-4, and will provide update.  Thank you!

Comment (valerymo) [reply bundles/rhsso-operator/bundles.yaml]: @MStokluska Fix done. Official registry bundle is used for 7.6.0-4. Attaching generated Index content
[index_rhsso7.6.0-4.txt](https://github.com/integr8ly/integreatly-operator/files/9399679/index_rhsso7.6.0-4.txt)
.
Thank you for review and comment

**Comment by MStokluska** [comment]
Verified the upgrade from 1.25.0 to 1.26.0 looks great.
Well done Valery, still looking at pipeline and delorean but we need to fix the manifests issue. I suggest, by removing the 18.0.0 entirely or removing the replaces from next manifests (NOTE: only do it for the manifests not the bundles themselves)

**Comment by MStokluska** [comment]
@valerymo I think this will need to be udpated as well https://github.com/integr8ly/integreatly-operator/blob/master/test/e2e/suite_test.go#L130 to rhsso-operator, as the deployment name changed from keycloak-operator to rhsso-operator

**Comment by MStokluska** [comment]
/lgtm

Good job Valery! :)

**Pull request merged**